optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/ClassBrowser.java

```diff
@@ -0,0 +1,55 @@
+package org.optaplanner.core.impl.domain.solution.descriptor;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.*;
+
+class ClassBrowser {
+
+    private static Set<Class<?>> getAllClassesInHierarchy(Class<?> root) {
+        Set<Class<?>> classes = new HashSet<>();
+        classes.add(root);
+        Class<?> superClass = root.getSuperclass();
+        while (superClass != Object.class) {
+            classes.add(superClass);
+            superClass = superClass.getSuperclass();
+        }
+        return classes;
+    }
+
+    /**
+     *
+     * @param root The class in which to start the search.
```

Comment:
suggestion: add either "never null" or "sometimes null" for params and return javadocs.

For example: `@param root Never null, the class in which to start the search.`